vmaas/reposcan/database/csaf_store.py

```diff
@@ -46,7 +48,129 @@ def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
+            if delete:
+                del self.csaf_product_map[(cpe, package, module)]
+            else:
+                self.csaf_product_map[(cpe, package, module)] = product_id
+
+    def _get_cve_rows(self, csaf_cves):
+        # TODO: Known_affected are only parsed as of now. Update here accordingly.
+        product_status_id = 4
+
+        to_insert = []
+        to_update = []
+        for cve, csaf_products in csaf_cves.items():
+            for csaf_product in csaf_products:
+                product_id = self.csaf_product_map.get(csaf_product.to_tuple())
+                if not product_id:
+                    self.logger.error(
+                        "CSAF product: %s associated with CVE: %s not found in the database.", csaf_product.to_tuple(), cve)
+                    continue
+                cve_in_cache = self.csaf_cve_map.get((cve, product_id))
+                if not cve_in_cache:
+                    to_insert.append((cve, product_id, product_status_id))
+                    # Insert key to the cache to avoid duplicates.
+                    # CVE ID will be populated after row is inserted into DB.
+                    self.csaf_cve_map[(cve, product_id)] = (-1, product_status_id)
+                else:
+                    cache_product_status_id = cve_in_cache[1]
+                    if product_status_id != cache_product_status_id:
+                        cve_id = cve_in_cache[0]
+                        to_update.append((cve_id, cve, product_id, product_status_id))
+
+        return to_insert, to_update
+
+    def _cve_refresh_cache(self, cur, delete=False):
+        for cve_id, cve, product_id, product_status_id in cur.fetchall():
+            if delete:
+                del self.csaf_cve_map[(cve, product_id)]
+            else:
+                self.csaf_cve_map[(cve, product_id)] = (cve_id, product_status_id)
+
+    def _populate_cves(self, csaf_cves):
+        self.csaf_cve_map = self._prepare_table_map(cols=["cve", "csaf_product_id"],
+                                                    to_cols=["id", "product_status_id"],
+                                                    table="csaf_cves")
+
+        self._populate_data(csaf_cves, "csaf_cves", ["cve", "csaf_product_id", "product_status_id"],
+                            self._get_cve_rows, self._cve_refresh_cache,
+                            "(%s::int, %s, %s::int, %s::int)")
+
+    def _populate_data(self, csaf_cves, table_name, cols, get_rows, refresh_cache, update_tmpl_str):
+        """Generic method to populate table with CSAF data."""
+        to_insert, to_update = get_rows(csaf_cves)
+        try:
+            cur = self.conn.cursor()
+            if to_insert:
+                execute_values(cur, f"""insert into {table_name} ({', '.join(cols)}) values %s
+                                        returning id, {', '.join(cols)}""",
+                               to_insert, page_size=len(to_insert))
```

Comment:
`page_size=len(to_insert)` this does not seem to have any effect since you are setting the page_size to total number of inserted items. It can be removed I think